src/views/List.jsx

```diff
@@ -1,15 +1,69 @@
+import { useState, useEffect } from 'react';
 import { ListItem } from '../components';
 
 export function List({ data }) {
+	// state for input
+	const [input, setInput] = useState('');
+	const [searchData, setSearchData] = useState(data);
+	const [isValid, setIsValid] = useState(false);
+	const [searchLength, setSearchLength] = useState(1);
+	// Handle event
+	const handleInputChange = (e) => {
+		setInput(e.target.value);
+	};
+
+	const handleInputClear = () => {
+		setInput('');
+	};
+
+	useEffect(() => {
+		const searchRegex = new RegExp(input, 'i');
+		const filteredData = data.filter((item) => searchRegex.test(item.name));
```

Comment:
```suggestion
		const filteredData = data.filter(item => item.name.includes(input.toLowerCase()));
```
an alternative to regex